TweetMate/src/main/java/com/ybsystem/tweetmate/usecases/StatusUseCase.java

```diff
@@ -0,0 +1,282 @@
+import static com.ybsystem.tweetmate.resources.ResString.STR_SUCCESS_UNLIKE;
+
+public class StatusUseCase {
+
+    private StatusUseCase() {
+    }
+
+    public static void retweet(TwitterStatus status) {
+        // Check
+        if (!status.isRetweeted() && !status.isPublic()) {
+            return;
+        }
+
+        // Create text
+        boolean isRetweeted = status.isRetweeted();
+        String confirm = isRetweeted ? STR_CONFIRM_UNRETWEET : STR_CONFIRM_RETWEET;
+        String success = isRetweeted ? STR_SUCCESS_UNRETWEET : STR_SUCCESS_RETWEET;
+
+        // Async
+        Observable<Object> observable = Observable.create(e -> {
+            try {
+                if (isRetweeted) {
+                    TweetMateApp.getTwitter().unRetweetStatus(status.getId());
+                } else {
+                    TweetMateApp.getTwitter().retweetStatus(status.getId());
+                }
+                e.onComplete();
+            } catch (TwitterException ex) {
+                e.onError(ex);
+            }
+        })
+        .subscribeOn(Schedulers.newThread())
+        .observeOn(AndroidSchedulers.mainThread());
+
+        // Result
+        DisposableObserver<Object> disp = new DisposableObserver<Object>() {
+            @Override
+            public void onNext(Object obj) {
+            }
+
+            @Override
+            public void onError(Throwable t) {
+                // Failed...
+                ToastUtils.showShortToast(ExceptionUtils.getErrorMessage(t));
+            }
+
+            @Override
+            public void onComplete() {
+                // Success
+                if (isRetweeted) {
+                    status.setRetweeted(false);
+                    status.setRetweetCount(status.getRetweetCount() - 1);
+                    boolean isDeleteTarget = status.isRetweet() && status.isMyTweet();
+                    EventBus.getDefault().post(isDeleteTarget ? new StatusEvent(status) : new StatusEvent());
+                } else {
+                    status.setRetweeted(true);
+                    status.setRetweetCount(status.getRetweetCount() + 1);
+                    EventBus.getDefault().post(new StatusEvent());
+                }
+                ToastUtils.showShortToast(success);
+            }
+        };
+
+        // Confirm
+        if (PrefSystem.getConfirmSettings().contains(RETWEET)) {
+            DialogUtils.showConfirm(
+                    confirm,
+                    (dialog, which) -> observable.subscribe(disp)
+            );
+        } else {
+            observable.subscribe(disp);
+        }
+    }
+
+    public static void favorite(TwitterStatus status) {
+        // Create text
+        boolean isFavorited = status.isFavorited();
+        boolean isLikeStyle = PrefAppearance.getLikeFavText().equals(STR_LIKE);
+        String confirm = isFavorited
+                ? (isLikeStyle ? STR_CONFIRM_UNLIKE : STR_CONFIRM_UNFAVORITE)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3358)